nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/cloudwatch/PutCloudWatchMetric.java

```diff
@@ -0,0 +1,203 @@
+            .name("Timestamp")
+            .displayName("Timestamp")
+            .description("A point in time expressed as the number of milliseconds since Jan 1, 1970 00:00:00 UTC. If not specified, the default value is set to the time the metric data was received")
+            .expressionLanguageSupported(true)
+            .required(false)
+            .addValidator(StandardValidators.LONG_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor UNIT = new PropertyDescriptor.Builder()
+            .name("Unit")
+            .displayName("Unit")
+            .description("The unit of the metric. (e.g Seconds, Bytes, Megabytes, Percent, Count,  Kilobytes/Second, Terabits/Second, Count/Second) For details see http://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricDatum.html")
+            .expressionLanguageSupported(true)
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final List<PropertyDescriptor> properties =
+            Collections.unmodifiableList(
+                    Arrays.asList(NAMESPACE, METRIC_NAME, VALUE, TIMESTAMP, UNIT, REGION, ACCESS_KEY, SECRET_KEY,
+                            CREDENTIALS_FILE, AWS_CREDENTIALS_PROVIDER_SERVICE, TIMEOUT, SSL_CONTEXT_SERVICE,
+                            ENDPOINT_OVERRIDE, PROXY_HOST, PROXY_HOST_PORT)
+            );
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    /**
+     * Create client using aws credentials provider. This is the preferred way for creating clients
+     */
+
+    @Override
+    protected AmazonCloudWatchClient createClient(ProcessContext processContext, AWSCredentialsProvider awsCredentialsProvider, ClientConfiguration clientConfiguration) {
+        getLogger().info("Creating client using aws credentials provider");
+        return new AmazonCloudWatchClient(awsCredentialsProvider, clientConfiguration);
+    }
+
+    /**
+     * Create client using AWSCredentials
+     *
+     * @deprecated use {@link #createClient(ProcessContext, AWSCredentialsProvider, ClientConfiguration)} instead
+     */
+    @Override
+    protected AmazonCloudWatchClient createClient(ProcessContext processContext, AWSCredentials awsCredentials, ClientConfiguration clientConfiguration) {
+        getLogger().debug("Creating client with aws credentials");
+        return new AmazonCloudWatchClient(awsCredentials, clientConfiguration);
+    }
+
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final AmazonCloudWatchClient client = getClient();
+
+        MetricDatum dataum = new MetricDatum();
```

Comment:
Should this be `datum` instead of `dataum`?  Admittedly not my most insightful review comment, I'm just trying to prove I'm paying attention.